### PR TITLE
Feature/inrel 5169 remove author header if spotlight content is present

### PR DESCRIPTION
**Pull request by mbrioski**
## [INREL-5169](https://jira.burda.com/browse/INREL-5169) 
from the ticket:
Please include a summary of the changes provided with this pull request and which issue has been As an SEO Manager I dont want to have a Spotlight Element + a Header on Author profiles. 
I want the same behaviour of the page like a term page (e.g. https://www.instyle.de/frisuren/flechtfrisuren) has: if there is no spotlight element a header is shown. however if there is a spotlight element the header is hidden.

## How to test:
Just add and remove a spotlight paragraph from an author and see if the spotlight paragraph is shown.
Basically the spotlight content has always the priority, if it is present then it must be shown and the header of the Author must disappear.

### CODE REVIEW

File: infinite.theme [hunk 1]
```diff
@@ -450,3 +450,39 @@ function _infinite_get_icon($node, $variables) {
...
+}
+
+/**
+ * Get the paragraph bundle
+ *
+ * @param \Drupal\user\Entity\User $user
+ * @param int|0 $paragraphPosition
+ *
+ * @return string
+ *
```
Comment (mgoedecke) [inline infinite.theme:479]: function throws InvalidArgumentException

File: infinite.theme [hunk 2]
```diff
@@ -450,3 +450,39 @@ function _infinite_get_icon($node, $variables) {
...
+
+/**
+ * Set infinite preprocess variables for User
+ *
+ * @param $variables
+ *
+ * @throws \Drupal\Core\TypedData\Exception\MissingDataException
+ */
+function infinite_preprocess_user(&$variables)  {
+  try {
```
Comment (mgoedecke) [inline infinite.theme:463]: you don't need ` ? true : false` here

File: templates/user/user--default.html.twig [hunk 3]
```diff
@@ -32,9 +32,9 @@
             <div class="item-author__profile-image">
                 {{ content.user_picture }}
             </div>
-            {% if content.has_header_block %}
```
Comment (mgoedecke) [inline templates/user/user--default.html.twig:35]: i think we should rename the variable to has_spotlight_paragraph

File: infinite.theme [hunk 4]
```diff
@@ -450,3 +450,39 @@ function _infinite_get_icon($node, $variables) {
...
+    $variables['has_spotlight_content'] = infinite_get_user_paragraph_bundle($variables['elements']['#user']) === 'spotlight' ? true : false;
+  }catch (InvalidArgumentException $ex) {
```
Comment (mgoedecke) [inline infinite.theme]: code style: missing a space after the curly bracket

**Review by mgoedecke — COMMENTED** [review]
Some minor adjustments, otherwise everything looks fine :)

File: infinite.theme (hunk 2 above)
Comment (mbrioski) [reply infinite.theme:463]: why not? i don't have a default value

File: infinite.theme (hunk 4 above)
Comment (mbrioski) [reply infinite.theme]: ok

File: infinite.theme (hunk 2 above)
Comment (mgoedecke) [reply infinite.theme:463]: Because the expression already returns a boolean.

$a === $b is the same as $a === $b ? true : false

Comment (mbrioski) [reply infinite.theme:463]: right :)
